src/d3a/d3a_core/simulation.py

```diff
@@ -374,10 +373,13 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
                 self.area.tick_and_dispatch()
                 self.area.update_area_current_tick()
                 if (self.simulation_config.external_connection_enabled and
-                        is_external_matching_enabled() and
-                        external_global_statistics.is_it_time_for_external_tick(
-                            current_tick_in_slot)):
-                    bid_offer_matcher.match_algorithm.publish_event_tick_myco()
+                        is_external_matching_enabled()):
+                    # If External matching is enabled, limit the number of ticks dispatched.
+                    if external_global_statistics.is_it_time_for_external_tick(
+                            current_tick_in_slot):
+                        bid_offer_matcher.event_tick()
+                else:
+                    bid_offer_matcher.event_tick()
```

Comment:
How about putting this into the `MycoMatcher.event_tick` so only `bid_offer_matcher.event_tick()` is left in here?